src/include/rados/librados.h

```diff
@@ -2007,6 +2034,34 @@ CEPH_RADOS_API int rados_aio_read(rados_ioctx_t io, const char *oid,
 		                  char *buf, size_t len, uint64_t off);
 
 /**
+ * Asychronously read data from an object
+ *
+ * The io context determines the snapshot to read from, if any was set
+ * by rados_ioctx_snap_set_read().
+ *
+ * The return value of the completion will be number of bytes read on
+ * success, negative error code on failure.
+ *
+ * @note only the 'complete' callback of the completion will be called.
+ *
+ * @param io the context in which to perform the read
+ * @param oid the name of the object to read from
+ * @param completion what to do when the read is complete
+ * @param buf where to store the results
+ * @param len the number of bytes to read
+ * @param off the offset to start reading from in the object
+ * @param info blkin trace information
+ * @returns 0 on success, negative error code on failure
+ */
+CEPH_RADOS_API int rados_aio_read_traced(rados_ioctx_t io,
```

Comment:
it'd be great to have tests using these new librados and librbd apis in test/librados/io.cc and test/librbd/test_librbd.cc (good place to show example blkin usage too)